src/openlcb/IfImpl.hxx

```diff
@@ -266,6 +266,78 @@ private:
     If::VNodeMap::Iterator it_;
 #endif
 };
+
+/// Message handler that is registered as a fallback handler in the interface's
+/// message dispatcher. This means that all incoming messages that were not
+/// matching the MTI / mask of any existing instantiated handler will end up
+/// here.
+///
+/// THe standard requires that when an addressed message is not handled by a
+/// node (e.g. because it does not know about the MTI at all), then an Optional
+/// Interaction Rejected reply be sent to the originator. This flow generates
+/// that OIR reply.
+class UnhandledAddressedMessageHandler : public IncomingMessageStateFlow
+{
+public:
+    UnhandledAddressedMessageHandler(If *service)
+        : IncomingMessageStateFlow(service)
+    {
+        iface()->dispatcher()->register_fallback_handler(this);
+    }
+
+    /// Handler callback for incoming messages.
+    Action entry() override
+    {
+        if (!message()->data()->dstNode)
+        {
+            // Destination is not a local node.
+            return release_and_exit();
+        }
+        return allocate_and_call(
+            iface()->addressed_message_write_flow(), STATE(fill_oir));
+    }
+
+    Action fill_oir()
+    {
+        auto rb = get_buffer_deleter(
+            get_allocation_result(iface()->addressed_message_write_flow()));
+        GenMessage *inm = message()->data();
+        GenMessage *outm = rb->data();
+        outm->reset(Defs::MTI_OPTIONAL_INTERACTION_REJECTED,
+            inm->dstNode->node_id(), inm->src,
+            error_payload(Defs::ERROR_UNIMPLEMENTED, inm->mti));
+        iface()->addressed_message_write_flow()->send(rb.release());
+        return release_and_exit();
+    }
+};
+
+/// This is an addressed message handler that catches errors (OIR and TDE) but
+/// ignores them. While this sounds not super useful, it ensures that these
+/// messages never trigger the UnhandledAddressedMessageHandler.
+class NullErrorHandler : public IncomingMessageStateFlow
```

Comment:
This does seem super not useful because the party that initiated the interaction will not be notified, it just times out. Should we have some way of notifying the initiator of the interaction? Perhaps at least file this as an enhancement request issue?

Reply:
I actually deleted this handler and replaced it with an if (mti != OIR && mti != TDE) in the other handler. This has the same result but takes less RAM and FLASH space.